### ISSUE TITLE
Infinite loop freezes up browser; doesn't display loop protector error

### ISSUE THREAD

**Issue by pamelafox** [Khan/live-editor#465]
Reported in ZenDesk.

Students code locks up browser (Chrome) instead of showing infinite loop error.

The code was:
```
background(219, 255, 255);

//roof
fill(174, 180, 214);
triangle(200, 28, 350, 150, 50, 150);
//actual house
fill(209, 181, 157);
rect(60, 150, 280, 207);
//door
fill(120, 80, 19);
rect(180, 280, 40, 77);
//window
fill(255, 255, 255);

for (var i = 90; i < 400; i+20){
    var x = 90;
    rect(x, 190, 50, 70);
    x += 100; 
}

//grass and short trees
var x = 0;
var grass = getImage("cute/GrassBlock");
var tree = getImage("cute/TreeShort");
while (x < 400) {
  image(grass, x, 303);
  image(tree, x, 288);
  x += 100; 
}
```

**Comment by kevinbarabash** [comment]
This is a really weird bug.  The loop detector works in this situation but you have to wait a really long time for the message to show up.  I tried a `for` loop with just a `debug` message, no problem.  It appears that `draw` commands are queued up by Chrome somewhere and after the execution stack finish running then all the draw commands are run.  The problem is that we queue up like a bazillion commands before the loop detector triggers.  Then we sit and wait while Chrome tries to draw everything.

**Comment by kevinbarabash** [comment]
I guess we could have a maximum number of draw calls before with trigger the loop protector triggers, just wrap all processing draw commands so that they update a counter somewhere and check that the counter doesn't exceed a certain amount.

**Comment by kevinbarabash** [comment]
@pamelafox I don't think this was a regression but rather us not testing with draw functions.  I'll see what I can do about this this weekend.

**Comment by DozenalFTW** [comment]
@ "I guess we could have a maximum number of draw calls before with trigger the loop protector triggers, just wrap all processing draw commands so that they update a counter somewhere and check that the counter doesn't exceed a certain amount."

Well, but that's no fun if you're doing something like `println(ellipse)` to go exploring... you'll find a reference to an `ellipse` that can't be explored. One of the things about JavaScript is that people can see lower-level code, and I'd hate to see that taken away.

**Comment by JavascriptFTW** [comment]
Has this bug been fixed? I tried to recreate it and the loop protector behaved normally.

**Comment by kevinbarabash** [comment]
@JavascriptFTW I believe this is a Chrome only behavior.  Regarding my Oct. 28th comment... the only thing I can think to do is to limit how many draw commands we do at a time.

**Comment by kevinbarabash** [comment]
There is a test in the test suite that's supposed to trigger the loop protector because it's drawing too much, but since all of the draw commands a queued and processed in a separate thread, the loop protector isn't trigger, but then when the draw commands are actually processed it locks up the browser.
